spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreResultsIterable.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core;
+
+import java.util.Iterator;
+
+import com.google.cloud.datastore.Cursor;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class DatastoreResultsIterable<T> implements Iterable<T> {
```

Comment:
Since Collection extends Iterable, I think just having DatastoreResultsCollection would be enough

Reply:
good catch!

Reply:
Actually we don't always want to have a collection and we need to return an iterator with cursor. So I think it make sense to keep both classes to avoid complex logic in case we want only an iterator.